auth/token-managers/token-manager.ts

```diff
@@ -0,0 +1,240 @@
+      // 1. request a new token
+      return this.pacedRequestToken().then(() => {
+        return this.tokenInfo[this.tokenName];
+      });
+    } else {
+      // If refresh needed, kick one off
+      if (this.tokenNeedsRefresh()) {
+        this.requestToken().then(tokenResponse => {
+          this.saveTokenInfo(tokenResponse);
+        });
+      }
+      // 2. use valid, managed token
+      return Promise.resolve(this.tokenInfo[this.tokenName]);
+    }
+  }
+
+  /**
+   * Setter for the disableSslVerification property.
+   *
+   * @param {boolean} value - the new value for the disableSslVerification
+   *   property
+   * @returns {void}
+   */
+  public setDisableSslVerification(value: boolean): void {
+    // if they try to pass in a non-boolean value,
+    // use the "truthy-ness" of the value
+    this.disableSslVerification = Boolean(value);
+  }
+
+  /**
+   * Set a completely new set of headers.
+   *
+   * @param {OutgoingHttpHeaders} headers - the new set of headers as an object
+   * @returns {void}
+   */
+  public setHeaders(headers: OutgoingHttpHeaders): void {
+    if (typeof headers !== 'object') {
+      // do nothing, for now
+      return;
+    }
+    this.headers = headers;
+  }
+
+  /**
+   * Paces requests to request_token.
+   *
+   * This method pseudo-serializes requests for an access_token
+   * when the current token is undefined or expired.
+   * The first caller to this method records its `requestTime` and
+   * then issues the token request. Subsequent callers will check the
+   * `requestTime` to see if a request is active (has been issued within
+   * the past 60 seconds), and if so will queue their promise for the
+   * active requestor to resolve when that request completes.
+   */
+  protected pacedRequestToken(): Promise<any> {
+    const currentTime = getCurrentTime();
+    if (this.requestTime > (currentTime - 60)) {
+      // token request is active -- queue the promise for this request
+      return new Promise((resolve, reject) => {
+        this.pendingRequests.push({resolve, reject});
+      });
+    } else {
+      this.requestTime = currentTime;
+      return this.requestToken().then(tokenResponse => {
+        this.saveTokenInfo(tokenResponse);
+        this.pendingRequests.forEach(({resolve}) => {
+          resolve();
+        });
+        this.pendingRequests = [];
+        this.requestTime = 0;
+      }).catch(err => {
+        this.pendingRequests.forEach(({reject}) => {
+          reject(err);
+        });
+        throw(err);
+      });
+    }
+  }
+
+  /**
+   * Request a token using an API endpoint.
+   *
+   * @returns {Promise}
+   */
+  protected requestToken(): Promise<any> {
+    const errMsg = '`requestToken` MUST be overridden by a subclass of JwtTokenManagerV1.';
+    const err = new Error(errMsg);
+    logger.error(errMsg);
+    return Promise.reject(err);
+  }
+
+  /**
+   * Save the token from the response and the calculated expiration time to the object's state.
```

Comment:
We should be explicit.  This method must set `accessToken` (see earlier comment), `expireTime`, and `refreshTime` based on the `tokenResponse`.

Reply:
Half fixed in e92c3de